drivers/serial/uart_intel_lw.c

```diff
@@ -0,0 +1,999 @@
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief UART driver for Intel FPGA UART Core IP
+ * Reference : Embedded Peripherals IP User Guide (22.3 and above): 55.
+ * Lightweight UART Core
+ */
+
+#define DT_DRV_COMPAT   intel_lw_uart
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/uart.h>
+
+#include <zephyr/drivers/serial/uart_intel_lw.h>
+
+#ifdef CONFIG_UART_LINE_CTRL
+	#ifndef CONFIG_UART_INTERRUPT_DRIVEN
+		/* flow control requires interrupt to work properly. */
+		#error "uart_intel_lw.c: Must enable UART_INTERRUPT_DRIVEN for line control!"
+	#endif
+#endif
```

Comment:
You can do `select UART_INTERRUPT_DRIVEN if UART_LINE_CTRL` in kconfig instead of build assertion.

Reply:
i have removed this check, as the driver select SERIAL_SUPPORT_INTERRUPT by default. So CONFIG_UART_INTERRUPT_DRIVEN is always enabled for this driver.